### DISCUSSION TITLE
qrcoder is getting too crowdy when adding a url

### DISCUSSION THREAD

**Discussion by krishna007-cloud** [Shane32/QRCoder#269] [Q&A] [answered]
Hi
I am trying to pass a "azure trigger function url2 as data to generate a qr code but it is getting too crowdy and unable to scan when i reduce the image size.

If i tried to use the same url on the-qrcode-generator.com it is looking much simpler. Is there a way that i can achieve the smaller qr so that i can easily scan when i reduce the image size . 
Tried using one of the renderers but unable to figure which one will be ideal for my situation.

Thanks

**Comment by nitrovent** [comment]
Have you tried reducing the error-correction level?

**Comment by codebude** [comment, marked as answer]
Hi @krish7129 ,

I guess with "too crowdy" you mean too much "black and white boxes", correct? Those boxes represent data. Too reduce the amount of boxes for the same amount of input data (=your url), you can reduce the amount of error correction data that is saved inside the QR code.

```csharp
QRCodeGenerator qrGenerator = new QRCodeGenerator();
string myUrl = "The payload aka the text which should be encoded.";
QRCodeData qrCodeData = qrGenerator.CreateQrCode(myUrl, QRCodeGenerator.ECCLevel.L);
```
By use of ECC Level => L less error correction data is coded into the QR code, which leads to less qr modules (=the black and white blocks). You can read more about the parameters here: https://github.com/codebude/QRCoder/wiki/How-to-use-QRCoder#parameters-of-the-createqrcode-method

Note: Keep in mind that error correction data is used, to make QR codes easier scannable for QR code readers. ECC data is the magic which allows QR codes to be scanned, even if parts of the graphic are covered or lost. So if you reduce the ECC data your QR code becomes "less robust". 
Note 2: Have you thought about taking your "long" url and passing it to an url shortener like https://bit.ly ? By encoding a shortened/shorter version of the url, you could also make the QR code "smaller".
